x-pack/plugins/security_solution/server/lib/detection_engine/rule_monitoring/logic/detection_engine_health/assets/install_assets_for_rule_monitoring.ts

```diff
@@ -79,20 +91,38 @@ const getAssetsForRuleMonitoring = (currentSpaceId: string) => {
   assetManagedTag.id = withSpaceId('fleet-managed');
   assetSecuritySolutionTag.id = withSpaceId('security-solution');
 
+  let tags: Array<SavedObjectsFindResult<TagAttributes>> | undefined;
+  try {
+    tags = await findTagsByName({
+      savedObjectsClient,
+      tagName: SECURITY_TAG_NAME,
+    });
+  } catch (err) {
+    // If no Security Solution tag exists, we will import it
+  }
+
+  const securityTagExists = tags && tags?.length > 0;
```

Comment:
The `getAssetsForRuleMonitoring` was not supposed to perform async operations, I think it should just return assets for importing, which should be a simple synchronous operation.

I'd like to suggest to implement it differently. In the `installAssetsForRuleMonitoring` function, before calling `getAssetsForRuleMonitoring`, I'd like to call a function that will install the security solution tag no matter what (regardless if it's already installed or not in the current space). This function could be called `ensureSecuritySolutionTagIsCreated` and would be imported from `x-pack/plugins/security_solution/server/lib/tags`. Ideally, the rule monitoring code shouldn't bother with passing any dependencies to this function, meaning the function should already have all the needed dependencies.